Correct row header width in widget.Table

Ju-B opened the pull request:
### Description:
Allows to set width and height of headers using methods: SetColumnWidth and SetRowHeight with index -1

Fixes #([4370](https://github.com/fyne-io/fyne/issues/4370))

### Checklist:
- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

andydotxyz commented:
Thanks for your contribution. I think there is some cross-branch issues here though, lots of unrelated commits have been pulled in.

Also does this not break the default header size? The new `headerSize` cache will default to 0,0 won't it?

Ju-B commented:
I'm not familiar at all with github and git in general, so I probably
messed up that part.
If you can advise how to edit this pull request, I'll do or I can close and
make a new one.

The default behaviour is not changed.
In createrenderer of Table and refresh of tablerenderer the cachesize is
and was set to minsize of header I added code to take into account if
height or width  has been specified.

 If no set is done, the default will be the header template's size as
before. This header behavior is coherent with the cells behavior.

An other implementation could avoid using the maps to store user specified
height or width but could rely on :
rowheaderwidth,columnheaderheight *int
Added to the Table type.
Benefit is that table.Minsize() will take the fast path if only the header
size is changed.
Do you have some thoughts?

Le ven. 24 nov. 2023, 23:17, Andy Williams ***@***.***> a
écrit :

> Thanks for your contribution. I think there is some cross-branch issues
> here though, lots of unrelated commits have been pulled in.
>
> Also does this not break the default header size? The new headerSize
> cache will default to 0,0 won't it?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/fyne-io/fyne/pull/4421#issuecomment-1826112318>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABNJDPUCIOMQC5O4SKV7BADYGEMJHAVCNFSM6AAAAAA7ZQNEJKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMRWGEYTEMZRHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
